server/src/main/java/org/elasticsearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -55,13 +58,21 @@
 public class TransportClusterStatsAction extends TransportNodesAction<ClusterStatsRequest, ClusterStatsResponse,
         TransportClusterStatsAction.ClusterStatsNodeRequest, ClusterStatsNodeResponse> {
 
+    private static final Logger logger = LogManager.getLogger(TransportClusterStatsAction.class);
+
     private static final CommonStatsFlags SHARD_STATS_FLAGS = new CommonStatsFlags(CommonStatsFlags.Flag.Docs, CommonStatsFlags.Flag.Store,
         CommonStatsFlags.Flag.FieldData, CommonStatsFlags.Flag.QueryCache,
         CommonStatsFlags.Flag.Completion, CommonStatsFlags.Flag.Segments);
 
     private final NodeService nodeService;
     private final IndicesService indicesService;
 
+    // guards #mappingStats, #analysisStats and #metaVersion
+    private final Object statsMutex = new Object();
+
+    private MappingStats mappingStats;
```

Comment:
Obviously this isn't sticky but in practice this will mostly be called on master by monitoring and the overhead of this is small. And even if this is less helpful for the stats REST API it's going to often help anyway statistically speaking I'd expect it to "often" help still.